src/WebJobs.Script/Binding/DocumentDBScriptBindingProvider.cs

```diff
@@ -89,25 +104,59 @@ public override Collection<Attribute> GetAttributes()
 
                 string databaseName = Context.GetMetadataValue<string>("databaseName");
                 string collectionName = Context.GetMetadataValue<string>("collectionName");
+                string connection = Context.GetMetadataValue<string>("connection");
 
-                DocumentDBAttribute attribute = null;
-                if (!string.IsNullOrEmpty(databaseName) || !string.IsNullOrEmpty(collectionName))
+                if (Context.IsTrigger)
                 {
-                    attribute = new DocumentDBAttribute(databaseName, collectionName);
+                    CosmosDBTriggerAttribute attributeTrigger = new CosmosDBTriggerAttribute(databaseName, collectionName);
+                    attributeTrigger.ConnectionString = connection;
+
+                    JObject leasesConfiguration = Context.GetMetadataValue<JObject>("lease");
```

Comment:
I know @mamaso recommended we bundle these into a single object, but I seem to recall that the best approach is to have the properties match 1:1 with the attribute in extensions. Eventually, that'll let you remove this code completely and we'll be able to auto-map them. If you change the shape of the metadata (as compared to the attribute), that mapping is harder.

@MikeStall may have more input here.